src/logic/metadata.ts

```diff
@@ -48,3 +49,22 @@ export async function listByIds(
 
     return { objects: paginatedData, pager: { page, total } };
 }
+
+export async function getDeletedObjects(
+    d2: D2,
+    originalParams?: Params
+): Promise<Pick<ModelCollection, "toArray" | "pager">> {
+    const { deletedObjects, pager } = (await axios.get(
+        d2.Api.getApi().baseUrl + "/deletedObjects",
+        {
+            withCredentials: true,
+            params: {
+                fields: ":all",
+                ...originalParams,
+                paging: false, // DHIS2 deletedObjects endpoint pager does not work properly
```

Comment:
That's sad. It's always returning the first page, right? I checked, the problem is here, the pager has a `total=0` that messes things up:

https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-api/src/main/java/org/hisp/dhis/deletedobject/DeletedObjectQuery.java#L178

It's still in dev, I guess we should report it...

Reply:
https://jira.dhis2.org/browse/DHIS2-7381